patternfly/patternfly-elements
# feat(text-input): validation attrs

## What I did

1. forward `pattern` to input
2. add `helper-text`
3. add `error-text`
4. add `validate-on`
5. add `--pf-c-form-control__error-text--m-status--Color` (NOT PRESENT in react version)
6. fix form state restore.
7. 
## Testing Instructions

1. check the demo
2. fill out a field and refresh - state should be restored.

## Notes to Reviewers

1. see http://v4-archive.patternfly.org/v4/components/text-input/design-guidelines#errors-and-validation
2. please critique my naming / api surface
